src/config.rs

```diff
@@ -341,9 +364,9 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
     // TODO: should we override server_ordering_strategy based on our IP support?
 
     let dns_proxy_addr: Address = match pc.proxy_metadata.get(DNS_PROXY_ADDR_METADATA) {
-        Some(dns_addr) => Address::new(ipv6_enabled, dns_addr)
+        Some(dns_addr) => Address::new(ipv6_localhost_enabled, dns_addr)
```

Comment:
How does this work if there's a mismatch between whether the zt pod has V6 disabled on `lo` and a random pod making use of DNS proxy doesn't, or vice-versa?

Reply:
Ugh. Forgot about that. Oddly, this fix seems to work in my environment, and the host has IPv6 but pods don't

Reply:
Oh, duh, ztunnel is not in the host.

Reply:
Ok I have fixed it but it comes with some complexity. Not too bad